lib/src/lib.rs

```diff
@@ -133,11 +133,12 @@ fn chunk_path_by_digest(repo_dir: &Path,
         DataType::Index => Path::new(config::INDEX_SUBDIR),
     };
 
+    let hex_digest = &digest.to_hex();
     repo_dir
         .join(i_or_c)
-        .join(&digest[0..1].to_hex())
-        .join(digest[1..2].to_hex())
-        .join(&digest.to_hex())
+        .join(&hex_digest[0..2])
```

Comment:
Before we were selecting 1 Byte and turning it into a hex which ends up being 2 characters.

If you look on line 136 you can see we change it to a string so now we select the first 2 characters for level one then the following 2 characters for level two.

The ends up translating to the same values as before with only a single to hex conversion instead of 3.

Here is a test from my machine proving that the paths are still the same, created a repo and stored my projects folder using the current master of rdedup. Then switch to this branch and stored the folder again and there are only a small amount of new chunks because of the branch switch but chunk paths are clearly resolving to the same values in both versions.

<img width="915" alt="screen shot 2017-05-26 at 14 54 50 1" src="https://cloud.githubusercontent.com/assets/382715/26508706/b5bc1d40-4223-11e7-87dc-ef1ec1056c47.png">